From eb3972008ef27f5228d8793168049acb93174932 Mon Sep 17 00:00:00 2001
From: Zhong Hongbo <hongbo.zhong@windriver.com>
Date: Thu, 15 Nov 2012 11:38:45 +0800
Subject: [PATCH 28/62] fsl_imx6: Add watchdog device init

Source: Extract from vendor-drop package, L3.0.15_12.04.01_ER_source.tar.gz
(May,2012).

Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/mach-mx6/Kconfig                     |    1 +
 arch/arm/mach-mx6/devices-imx6q.h             |    4 ++++
 arch/arm/mach-mx6/mach-mx6q_sabrelite.c       |    2 ++
 arch/arm/plat-mxc/devices/platform-imx2-wdt.c |    9 +++++++++
 4 files changed, 16 insertions(+)

diff --git a/arch/arm/mach-mx6/Kconfig b/arch/arm/mach-mx6/Kconfig
index 897b666..d8d6baa 100644
--- a/arch/arm/mach-mx6/Kconfig
+++ b/arch/arm/mach-mx6/Kconfig
@@ -28,6 +28,7 @@ config MACH_MX6Q_SABRELITE
 	select IMX_HAVE_PLATFORM_MXC_EHCI
 	select IMX_HAVE_PLATFORM_FSL_OTG
 	select IMX_HAVE_PLATFORM_FSL_USB_WAKEUP
+	select IMX_HAVE_PLATFORM_IMX2_WDT
 	help
 	  Include support for i.MX 6Quad SABRE Lite platform. This includes specific
 	  configurations for the board and its peripherals.
diff --git a/arch/arm/mach-mx6/devices-imx6q.h b/arch/arm/mach-mx6/devices-imx6q.h
index 3b732f9..0542279 100644
--- a/arch/arm/mach-mx6/devices-imx6q.h
+++ b/arch/arm/mach-mx6/devices-imx6q.h
@@ -89,4 +89,8 @@ extern const struct imx_hdmi_soc_data imx6q_imx_hdmi_soc_dai_data __initconst;
 	imx_add_hdmi_soc_dai(&imx6q_imx_hdmi_soc_dai_data)
 
 extern struct platform_device *__init imx6q_add_ahci_imx(void);
+
+extern const struct imx_imx2_wdt_data imx6q_imx2_wdt_data[] __initconst;
+#define imx6q_add_imx2_wdt(id, pdata)   \
+	imx_add_imx2_wdt(&imx6q_imx2_wdt_data[id])
 #endif
diff --git a/arch/arm/mach-mx6/mach-mx6q_sabrelite.c b/arch/arm/mach-mx6/mach-mx6q_sabrelite.c
index 93a88a5..845e263 100644
--- a/arch/arm/mach-mx6/mach-mx6q_sabrelite.c
+++ b/arch/arm/mach-mx6/mach-mx6q_sabrelite.c
@@ -620,6 +620,8 @@ static void __init mx6_sabrelite_board_init(void)
 	imx6q_add_hdmi_soc();
 	imx6q_add_hdmi_soc_dai();
 
+	imx6q_add_imx2_wdt(0, NULL);
+
 	/* SD/MMC */
 	platform_device_register(&sabrelite_vmmc_reg_devices);
 	imx6q_add_sdhci_usdhc_imx(3, &mx6q_sabrelite_sd4_data);
diff --git a/arch/arm/plat-mxc/devices/platform-imx2-wdt.c b/arch/arm/plat-mxc/devices/platform-imx2-wdt.c
index 5e07ef2..872515a 100644
--- a/arch/arm/plat-mxc/devices/platform-imx2-wdt.c
+++ b/arch/arm/plat-mxc/devices/platform-imx2-wdt.c
@@ -62,6 +62,15 @@ const struct imx_imx2_wdt_data imx53_imx2_wdt_data[] __initconst = {
 };
 #endif /* ifdef CONFIG_SOC_IMX53 */
 
+#ifdef CONFIG_SOC_IMX6Q
+const struct imx_imx2_wdt_data imx6q_imx2_wdt_data[] __initconst = {
+#define imx6q_imx2_wdt_data_entry(_id, _hwid)                           \
+	imx_imx2_wdt_data_entry(MX6Q, _id, _hwid, SZ_16K)
+	imx6q_imx2_wdt_data_entry(0, 1),
+	imx6q_imx2_wdt_data_entry(1, 2),
+};
+#endif
+
 struct platform_device *__init imx_add_imx2_wdt(
 		const struct imx_imx2_wdt_data *data)
 {
-- 
1.7.9.7

